### PR TITLE
CRM-21344 - Fix documentation links in installer error messages

### PR DESCRIPTION
**Pull request by wmortada**
Overview
----------------------------------------
The installer produces error messages if anything goes wrong.  These messages include a helpful link to documentation.  Unfortunately this link is broken so you can't click on the link.   This PR fixes this so the links work correctly.

Before
----------------------------------------
If you click on the link to documentation in the error message you get a page not found error.

After
----------------------------------------
If you click on the link to documentation you go to the relevant page in the user guide.

Technical Details
----------------------------------------
Basically the code expects a URL but gets back HTML including the <a> tag.  I've changed the code so that it expects a fully formed link in HTML.

I've also simplified the code to remove a redundant check for the existence of the ts() function.  If ts() isn't defined then nor will docURL2().

---

 * [CRM-21344: Links to documentation from installer error messages are broken](https://issues.civicrm.org/jira/browse/CRM-21344)

### CODE REVIEW

**Comment by wmortada** [comment]
@totten I spotted this bug when working on [CRM-21297](https://issues.civicrm.org/jira/browse/CRM-21297)

File: install/index.php [hunk 1]
```diff
@@ -223,7 +223,7 @@
   }
 
   $docLink = CRM_Utils_System::docURL2('Installation and Upgrades', FALSE, ts('Installation Guide'), NULL, NULL, "wiki");
-  $errorMsg = ts("CiviCRM has already been installed. <ul><li>To <strong>start over</strong>, you must delete or rename the existing CiviCRM settings file - <strong>civicrm.settings.php</strong> - from <strong>%1</strong>.</li><li>To <strong>upgrade an existing installation</strong>, <a href='%2'>refer to the online documentation</a>.</li></ul>", array(1 => $settings_directory, 2 => $docLink));
+  $errorMsg = ts("CiviCRM has already been installed. <ul><li>To <strong>start over</strong>, you must delete or rename the existing CiviCRM settings file - <strong>civicrm.settings.php</strong> - from <strong>%1</strong>.</li><li>To <strong>upgrade an existing installation</strong>, refer to the online documentation: %2.</li></ul>", array(1 => $settings_directory, 2 => $docLink));
```
Comment (totten) [inline install/index.php]: This works well.

File: install/index.php [hunk 2]
```diff
@@ -1842,13 +1842,7 @@ function getSiteDir($cmsPath, $str) {
...
   if ($showRefer) {
     $docLink = CRM_Utils_System::docURL2('Installation and Upgrades', FALSE, 'Installation Guide', NULL, NULL, "wiki");
-
-    if (function_exists('ts')) {
-      $errorMsg .= '<p>' . ts("<a %1>Refer to the online documentation for more information</a>", array(1 => "href='$docLink'")) . '</p>';
-    }
-    else {
-      $errorMsg .= '<p>' . sprintf("<a %s>Refer to the online documentation for more information</a>", "href='$docLink'") . '</p>';
-    }
+    $errorMsg .= '<p>' . ts("Refer to the online documentation for more information: ") . $docLink . '</p>';
```
Comment (totten) [inline install/index.php]: @wmortada Do you have a specific reason for removing the `function_exists('ts')` guard -- or just a general sense that it feels redundant?

Without a more specific reason, I'm inclined to keep the guard in there --the `install/index.php` is general a preboot environment where we can't take much for granted, and (as an error handler) it makes even more sense for `errorDisplayPage()` to be conscientious about preconditions.

Comment (totten) [reply install/index.php]: Ah, that's a good point.

CC @bgm -- any thoughts from i18n perspective? Are you persuaded too?

**Comment by wmortada** [comment]
I've added some code to check that the functions are defined.

File: install/index.php [hunk 3]
```diff
@@ -1840,14 +1840,18 @@ function getSiteDir($cmsPath, $str) {
...
+  // Add a link to the documentation
   if ($showRefer) {
-    $docLink = CRM_Utils_System::docURL2('Installation and Upgrades', FALSE, 'Installation Guide', NULL, NULL, "wiki");
+    if (is_callable(array('CRM_Utils_System', 'docURL2'))) {
+      $docLink = CRM_Utils_System::docURL2('Installation and Upgrades', FALSE, 'Installation Guide', NULL, NULL, "wiki");
+    }
 
     if (function_exists('ts')) {
-      $errorMsg .= '<p>' . ts("<a %1>Refer to the online documentation for more information</a>", array(1 => "href='$docLink'")) . '</p>';
+      $errorMsg .= '<p>' . ts("Refer to the online documentation for more information: ") . $docLink . '</p>';
```
Comment (seamuslee001) [inline install/index.php]: just thinking here @totten @wmortada if we don't have docURL2 then will this potentially create a unset variable issue? would it make more sens say around L1843 todo 
```php
$docLink = '';
```

Comment (wmortada) [reply install/index.php]: @seamuslee001 good point.  I've added some code to fix this.

**Comment by eileenmcnaughton** [comment]
@seamuslee001 @wmortada - is this good to merge. Looks like a trivial low-risk PR that fixes an identifiable issue & comments seem to be addressed so I will merge on your 'yay'

**Comment by seamuslee001** [comment]
@eileenmcnaughton i'm good with this

**Comment by eileenmcnaughton** [comment]
Sold - to the man on the left

**Comment by wmortada** [comment]
Thanks @eileenmcnaughton!
